### ISSUE TITLE
Windows installer does not work behind proxy

### ISSUE THREAD

**Issue by faph** [OpenHydrology/OH-Auto-Statistical#7]
The Windows installer uses conda to install the application packages. Conda needs proxy details for it to work behind a proxy.

**Comment by faph** [comment]
Conda uses `requests` which is supposed to be able to pick up the proxy settings from the Windows registry. It seems that there may be some issues with some complex (authenticating?) proxies?

**Comment by neilnutt** [comment]
Yes it would be an authentication proxy as to use the qgis repos I need to
supply my username and password. If that is the best qgis can do I suspect
that might be reasonable to set manually entering as a best achievable
scenario.

**Comment by faph** [comment]
Thanks. Are you able to figure out where QGIS stores the username and password?

**Comment by faph** [comment]
Found it. It saves it in plain text in the Windows registry. I'm assuming your proxy password is the same as your Windows/domain logon? Are your sysadmins happy that QGIS stores this in plain text?

**Comment by faph** [comment]
It would be useful if you're able to run the `urllib` and `requests` `getproxies()` methods to see whether python detects the proxy address itself.

Note that all of this equally applies to downloading NRFA data, not just the installer.

**Comment by neilnutt** [comment]
I'll take a look tomorrow

**Comment by faph** [comment]
It seems that QGIS is using some QT libraries for network access (http://qt-project.org/doc/qt-4.8/qnetworkproxy.html). In python it doesn't seem very straightforward dealing with NTLM proxies... It would be worth seeing if `urllib` gets through if you supplied it with your credentials. The problem of course is that I can't test any of this.

**Comment by neilnutt** [comment]
My work proxy is pretty complex and possibly not typical. I think it would
just be best to neatly catch the failure, tell the user what has gone wrong
and suggest trying outside the proxy/firewall.

**Comment by faph** [comment]
It's likely that your company's proxy is Microsoft NTLM proxy (which seems to be a non-standard beast). Unfortunately there are probably quite a few big companies out there who use it.

**Comment by faph** [comment]
For those amongst us that don't mind storing their proxy details in user environment variables, as a work-around please set these _before_ installing OH Auto Statistical:

`HTTP_PROXY`: `http://user:password@server:port`
`HTTPS_PROXY`: `http://user:password@server:port`

Not sure if that works on all proxies.

**Comment by neilnutt** [comment]
Will try this on Monday.

**Comment by faph** [comment]
See also:
https://github.com/pypa/pip/issues/1182
https://github.com/kennethreitz/requests/issues/2036

One option we can try is to install the [Firewall Client for ISA Server](http://www.microsoft.com/en-gb/download/confirmation.aspx?id=10193), unset (system-wide) environment variables `HTTP_PROXY` and `HTTPS_PROXY` (if set) and set `NO_PROXY=1`.

We can easily integrate this in the installer, but if someone behind an MS proxy can first tell me that this actually works, that would be great. I can't test it.
